[Bug]: [Nightly] Search iterator invalid partition test fails on describe_collection num_partitions check

## Environment

- **Milvus Version**: unknown-20260513-bd828e3ff8 / commit bd828e3ff8
- **Deployment Mode**: cluster
- **MQ**: pulsar-mmap, woodpecker, woodpecker-service
- **SDK**: pymilvus 2.7.0rc214
- **OS**: Jenkins 4AM E2E environment

## Reproduction

### Option B: Steps

1. Run master nightly from DAILY-CI-TRIGGER build 209, master dispatcher build 10250.
2. Check the failed downstream E2E builds:
   - distributed-pulsar-mmap: https://jenkins-milvus-ci.milvus.io/job/MILVUS-CI-V2-PR-PIPELINES/job/milvus-e2e-pipeline-4am/22826/consoleText
   - distributed-woodpecker: https://jenkins-milvus-ci.milvus.io/job/MILVUS-CI-V2-PR-PIPELINES/job/milvus-e2e-pipeline-4am/22829/consoleText
   - distributed-woodpecker-service: https://jenkins-milvus-ci.milvus.io/job/MILVUS-CI-V2-PR-PIPELINES/job/milvus-e2e-pipeline-4am/22832/consoleText
3. Observe the failure in `milvus_client/test_milvus_client_search_iterator.py::TestMilvusClientSearchIteratorInValid::test_milvus_client_search_iterator_with_invalid_partition_name[client_partition_85Jv3Pf3]`.

### Trigger Conditions

- **Frequency**: observed in all three master deployment modes in this nightly run
- **First observed after**: master commit bd828e3ff8 in DAILY-CI-TRIGGER build 209
- **Does NOT happen when**: not confirmed

## Expected Behavior

The test should create the collection and partition successfully, then call `search_iterator` with an invalid `partition_names` value and verify the expected invalid partition-name error.

## Actual Behavior

The test fails before reaching the invalid `search_iterator` call. After `create_partition`, it calls `describe_collection` and expects `num_partitions` to be `2`, but the property check fails.

## Error Logs

```text
FAILED milvus_client/test_milvus_client_search_iterator.py::TestMilvusClientSearchIteratorInValid::test_milvus_client_search_iterator_with_invalid_partition_name[client_partition_85Jv3Pf3] - AssertionError
```

Representative traceback from build 22826:

```text
self.create_collection(client, collection_name, default_dim, consistency_level="Bounded")
self.create_partition(client, collection_name, partition_name)
collections = self.list_collections(client)[0]
assert collection_name in collections
>       self.describe_collection(client, collection_name,
                                 check_task=CheckTasks.check_describe_collection_property,
                                 check_items={"collection_name": collection_name,
                                              "dim": default_dim,
                                              "consistency_level": 2,
                                              "num_partitions": 2})

check_items = {'collection_name': 'milvus_client_api_search_iterator_2D8OQu8R', 'consistency_level': 2, 'dim': 128, 'num_partitions': 2}

>       assert res["num_partitions"] == check_items.get("num_partitions", 1)
E       AssertionError

/home/jenkins/agent/workspace/MILVUS-CI-V2-PR-PIPELINES/milvus-e2e-pipeline-4am/tests/python_client/check/func_check.py:282: AssertionError
```

## Non-default Configuration

```yaml
N/A
```

## Analysis Hints (Optional)

This failure blocks the intended invalid partition-name verification. Please check whether `describe_collection` should report `num_partitions` as `2` after `create_partition`, or whether this test should not assert `num_partitions` for normal partitions.

--- a/internal/proxy/meta_cache.go
+++ b/internal/proxy/meta_cache.go
@@ -1225,6 +1225,7 @@ func (m *MetaCache) RemovePartition(ctx context.Context, database string, collec
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	staleCollectionInfo := m.shouldStaleCollectionInfoLocked(collectionID, version)
 	names := make(map[string]struct{})
 	realNames := make(map[string]struct{})
 	for _, dbName := range partitionInvalidationDatabases(database) {
@@ -1249,11 +1250,24 @@ func (m *MetaCache) RemovePartition(ctx context.Context, database string, collec
 			}
 		}
 
+		if staleCollectionInfo {
+			// Partition DDL changes collection-level fields such as NumPartitions,
+			// without requiring removeCollectionByID's partition-cache sweep.
+			for name := range names {
+				m.staleCollectionInfoByNameLocked(dbName, name)
+			}
+			if collectionID != 0 {
+				m.sfGlobal.Forget(buildSfKeyById(dbName, collectionID))
+			}
+		}
 		for name := range names {
 			m.stalePartitionCacheLocked(dbName, name, partitionName, version)
 		}
 		clear(names)
 		clear(realNames)
+	}
+	if staleCollectionInfo && collectionID != 0 && version != 0 {
+		m.collectionCacheVersion[collectionID] = version
 	}
 
 	log.Ctx(ctx).Debug("remove partition", zap.String("db", database), zap.Int64("collectionID", collectionID), zap.String("collection", collectionName), zap.String("partition", partitionName), zap.Uint64("version", version))
@@ -1363,6 +1377,21 @@ func (m *MetaCache) stalePartitionCacheLocked(database, collectionName, partitio
 	m.partitionCache.Stale(buildPartitionSfKey(database, collectionName, partitionName), version)
 	m.sfCollLevelPartitionCache.Forget(collectionKey)
 	m.collLevelPartitionCache.Stale(collectionKey, version)
+}
+
+func (m *MetaCache) shouldStaleCollectionInfoLocked(collectionID UniqueID, version uint64) bool {
+	if collectionID == 0 || version == 0 {
+		return true
+	}
+	return m.collectionCacheVersion[collectionID] <= version
+}
+
+func (m *MetaCache) staleCollectionInfoByNameLocked(database, collectionName string) {
+	if db, ok := m.collInfo[database]; ok {
+		delete(db, collectionName)
+	}
+
+	m.sfGlobal.Forget(buildSfKeyByName(database, collectionName))
 }
 
 func (m *MetaCache) backgroundGCLoop(stopCh <-chan struct{}) {

--- a/internal/proxy/meta_cache_test.go
+++ b/internal/proxy/meta_cache_test.go
@@ -2356,6 +2356,93 @@ func TestMetaCache_RemovePartition(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestMetaCache_RemovePartitionInvalidatesCollectionInfo(t *testing.T) {
+	ctx := context.Background()
+	rootCoord := mocks.NewMockMixCoordClient(t)
+
+	rootCoord.EXPECT().ListPolicy(mock.Anything, mock.Anything).Return(&internalpb.ListPolicyResponse{
+		Status: merr.Success(),
+	}, nil).Maybe()
+
+	describeCalls := 0
+	rootCoord.EXPECT().DescribeCollection(mock.Anything, mock.Anything).RunAndReturn(
+		func(ctx context.Context, req *milvuspb.DescribeCollectionRequest, opts ...grpc.CallOption) (*milvuspb.DescribeCollectionResponse, error) {
+			describeCalls++
+			numPartitions := int64(1)
+			requestTime := uint64(1000)
+			if describeCalls > 1 {
+				numPartitions = 2
+				requestTime = 3000
+			}
+			return &milvuspb.DescribeCollectionResponse{
+				Status:        merr.Success(),
+				CollectionID:  1,
+				DbName:        "db",
+				NumPartitions: numPartitions,
+				RequestTime:   requestTime,
+				Schema: &schemapb.CollectionSchema{
+					Name:   "collection",
+					Fields: []*schemapb.FieldSchema{},
+				},
+			}, nil
+		}).Times(2)
+
+	cache, err := NewMetaCache(rootCoord)
+	assert.NoError(t, err)
+	defer cache.Close()
+
+	info, err := cache.GetCollectionInfo(ctx, "db", "collection", 1)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(1), info.numPartitions)
+
+	cache.RemovePartition(ctx, "db", 1, "collection", "par1", 2000)
+
+	info, err = cache.GetCollectionInfo(ctx, "db", "collection", 1)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(2), info.numPartitions)
+}
+
+func TestMetaCache_RemovePartitionKeepsCollectionInfoForStaleVersion(t *testing.T) {
+	ctx := context.Background()
+	rootCoord := mocks.NewMockMixCoordClient(t)
+
+	rootCoord.EXPECT().ListPolicy(mock.Anything, mock.Anything).Return(&internalpb.ListPolicyResponse{
+		Status: merr.Success(),
+	}, nil).Maybe()
+
+	describeCalls := 0
+	rootCoord.EXPECT().DescribeCollection(mock.Anything, mock.Anything).RunAndReturn(
+		func(ctx context.Context, req *milvuspb.DescribeCollectionRequest, opts ...grpc.CallOption) (*milvuspb.DescribeCollectionResponse, error) {
+			describeCalls++
+			return &milvuspb.DescribeCollectionResponse{
+				Status:        merr.Success(),
+				CollectionID:  1,
+				DbName:        "db",
+				NumPartitions: 1,
+				RequestTime:   1000,
+				Schema: &schemapb.CollectionSchema{
+					Name:   "collection",
+					Fields: []*schemapb.FieldSchema{},
+				},
+			}, nil
+		}).Times(1)
+
+	cache, err := NewMetaCache(rootCoord)
+	assert.NoError(t, err)
+	defer cache.Close()
+
+	info, err := cache.GetCollectionInfo(ctx, "db", "collection", 1)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(1), info.numPartitions)
+
+	cache.RemovePartition(ctx, "db", 1, "collection", "par1", 500)
+
+	info, err = cache.GetCollectionInfo(ctx, "db", "collection", 1)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(1), info.numPartitions)
+	assert.Equal(t, 1, describeCalls)
+}
+
 func TestMetaCache_PartitionCache_Concurrent(t *testing.T) {
 	ctx := context.Background()
 	rootCoord := mocks.NewMockMixCoordClient(t)

